apps/shelter/src/pages/volunteers/search/_hooks/useRecruitmentSearch.ts

```diff
@@ -0,0 +1,25 @@
+import { ChangeEvent } from 'react';
+import { useSearchFilter } from 'shared/hooks/useSearchFilter';
+import { useSearchKeyword } from 'shared/hooks/useSearchKeyword';
+
+import { SearchFilter } from '@/pages/volunteers/search/_types/filter';
+
+export const useRecruitmentSearch = () => {
+  const [searchFilter, setSearchFilter] = useSearchFilter<SearchFilter>();
+
+  const setKeywordFilter = (keyword: string) => setSearchFilter({ keyword });
+
+  useSearchKeyword(setKeywordFilter);
+
+  const handleChangeSearchFilter = (event: ChangeEvent<HTMLSelectElement>) => {
+    const { name, value } = event.target;
```

Comment:
p5: 저는 이렇게 다 꺼내서 사용하는 걸 좋아합니다! 이부분은 취향 같네요
```suggestion
  const handleChangeSearchFilter = ({target:{name,value}}:ChangeEvent<HTMLSelectElement>) => {
```